packages/flutter/lib/src/material/theme.dart

```diff
@@ -34,14 +36,22 @@ class Theme extends InheritedWidget {
   /// Specifies the color and typography values for descendant widgets.
   final ThemeData data;
 
+  /// True if this theme shadows the one created by the [MaterialApp].
+  final bool isShadowTheme;
+
   static final ThemeData _kFallbackTheme = new ThemeData.fallback();
 
   /// The data from the closest instance of this class that encloses the given context.
   ///
   /// Defaults to the fallback theme data if none exists.
-  static ThemeData of(BuildContext context) {
-    Theme theme = context.inheritFromWidgetOfExactType(Theme);
-    return theme?.data ?? _kFallbackTheme;
+  ///
+  /// If [shadowTheme] is true and the closest Theme ancestor was installed by
+  /// the [MaterialApp] - in other words if the closest Theme ancestor does not
+  /// shadow the app's theme - then return null.
+  static ThemeData of(BuildContext context, { bool shadowTheme: false }) {
```

Comment:
maybe shadowThemeOnly? Or `secondaryThemeOnly` if you call the other argument `isPrimaryTheme`?